src/apps/backend/modules/config/internals/config_file.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+
+import os
+from pathlib import Path
+from typing import Any, Optional
+
+import yaml
+
+
+class ConfigFile:
+    """
+    Represents a configuration file that can be loaded, merged, and updated with environment variables.
+    """
+
+    def __init__(self, filename: str, config_path: Path):
+        self.filename = filename
+        self.config_path = config_path
+        self.content: dict[str, Any] = {}
+
+    def load(self) -> None:
+        """
+        Loads the content of the configuration file.
+        """
+        file_path = self.config_path / self.filename
```

Comment:
though `/` might work, this is usually not a good practice as file path concatenation string can often differ from underlying platform. Can we look into recommended way to construct paths in Python?